vyper/codegen/types/types.py

```diff
@@ -397,12 +397,16 @@ def FAIL():
                 custom_structs,
             )
 
+        elif "type" in item._metadata:
```

Comment:
`parse_type` is supposed to be independent of the new type checking / annotation machinery. it's a good check that `parse_type` matches the type generated by the new type checker. if it doesn't match, it's indicative of a deeper problem and shouldn't be papered over like this.